### ISSUE TITLE
EndpointResourceDescription.Protocol flexibility

### ISSUE THREAD

**Issue by Jm-Fox** [Azure/service-fabric-issues#179]
EndpointResourceDescription.Protocol is of enum type EndpointProtocol, which only supports Http, Https, and Tcp.

If someone wants to use UDP, or anything else, it can't be specified in the EndpointProtocol.

I'm presently modifying @suremaker's RemoteExecution library to support Service Fabric, among other things. I would like to integrate his protocol resolving with Service Fabric's endpoint description protocol.

Here I would much rather have "{description.Protocol}" than "net":
https://github.com/Jm-Fox/RemoteExecution-Net/blob/master/RemoteExecution.ServiceFabric/CommonHelper.cs#L20

Long story short, it would be great if EndpointResourceDescription.Protocol to be of type string, not enum.

**Comment by masnider** [comment]
Hi @Jm-Fox - right now you're correct in that those are the only values that are present, however they aren't the only protocols that are supported. Ex: if I wanted a service to be able to advertise an amqp endpoint, it could select TCP in the endpoint description, or if I wanted to use websockets I'd select HTTP. 

This selector really only controls what the Service Fabric hosting layer does in terms of setting up resources on the machine. Ex: when you select TCP is opens up ports in the firewall for UDP and TCP for the process that gets created. If you select http it adds urlacls for those ports that are assigned or provided. This doesn't really have anything to do with the protocol that the service actually ends up using to communicate. 

We realize this is unclear and we will improve both the documentation and the examples to make it more clear, as well as consider making the protocol list more expansive so that people don't feel like there's a limit here. 

In the meantime, did you try TCP and get things working? You should be able to. Please let us know if you can give that a try and let us know if you have any problems.
